web-console/src/views/segments-view/segments-view.tsx

```diff
@@ -315,9 +346,16 @@ export class SegmentsView extends React.PureComponent<SegmentsViewProps, Segment
         onFilteredChange={(filtered, column) => {
           this.setState({ segmentFilter: filtered });
         }}
-        onFetchData={noSqlMode ? this.fecthClientSideData : this.fetchData}
+        onFetchData={
+          noSqlMode
+            ? this.fetchClientSideData
+            : groupSegmentsBy
+            ? this.fetchGroupData
+            : this.fetchData
```

Comment:
you should just have one fetch data function and it should condition on `groupSegmentsBy`